README.md

```diff
@@ -739,6 +739,7 @@ The OpenTelemetry Operator _might_ work on versions outside of the given range,
 
 | OpenTelemetry Operator | Kubernetes     | Cert-Manager | Prometheus-Operator |
 |------------------------|----------------| ------------ |---------------------|
+| v0.106.0               | v1.23 to v1.30 | v1           | v0.74.0             |
```

Comment:
[prom-operator is v0.75.0 now](https://github.com/open-telemetry/opentelemetry-operator/blob/main/go.mod#L23)

Reply:
done and rebased with #3200